File: longestCommonPrefix/longestCommonPrefix.js
/*
@param {string[]} strs
@return {string}
*/

 var longestCommonPrefix = function(strs) {
  //set prefix to first word (split into array)
  var prefix = strs[0].split('');
  // loop through second word and check each letter to see if it matches the prefix
  for (var i = 1; i < strs.length; i ++) {
      var curr = strs[i].split('');
      var temp = [];
      for (var j = 0; j < curr.length; j ++) {
          // add the matching letters to new prefix
          if (curr[j] === prefix[j]) {
              temp.push(curr[j]);
          // cut off loop once letters no longer match
          } else {
              break;
          }
      }
      prefix = temp;
  }
  return prefix.join('');
};

export default longestCommonPrefix;

File: general/longestCommonPrefix/longestCommonPrefix.py
def longest_common_prefix(strs):
  prefix = list(strs[0])
  i = 1
  while i < len(strs):
    curr = list(strs[i])
    temp = []
    j = 0
    ## set length variable to be the shortest length of curr and prefix
    l = len(curr) if len(curr) < len(prefix) else len(prefix)
    while j < l-1:
      if curr[j] == prefix[j]:
        temp.append(curr[j])
      else:
        break
      j += 1
    prefix = temp
    i += 1
  separator = ''
  return separator.join(prefix)


print(longest_common_prefix(["flower", "flow", "flight"])) # should be fl
print(longest_common_prefix(["look", "loop", "loose"])) # should be lo